src/engine/controls/loopingcontrol.cpp

```diff
@@ -958,6 +960,10 @@ void LoopingControl::updateBeatLoopingControls() {
 }
 
 void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint, bool enable) {
+
+    // if a seek was queued in the engine buffer wait until it is executed
+    while(getEngineBuffer()->isSeekQueued()){}
```

Comment:
This doesn't look sane to me. A busy wait on a condition that is controlled by another thread?

Reply:
I expected something like this but was not sure which other mechanism to use. Just tell me what would me the right one and I will adapt my solution.

Reply:
A straight forward solution is to predict the effect of the scheduled seek here.

Reply:
I think you need to make the 
LoopSamples loopSamples = m_loopSamples.getValue(); conditional. 
If there is a cued seek, you need to use m_queuedSeekPosition.getValue(). 

By luck you can ignore the quantize flag. If not you need to adjust the loop along with the   m_queuedSeekPosition.